serde-support/src/main/java/io/micronaut/serde/support/serdes/InstantSerde.java

```diff
@@ -27,8 +28,7 @@
  *
  * @since 1.0.0
  */
-@Singleton
-public final class InstantSerde extends NumericSupportTemporalSerde<Instant> implements TemporalSerde<Instant> {
+public final class InstantSerde extends NumericSupportTemporalSerde<Instant> implements TemporalSerde<Instant>, SerdeRegistrar<Instant> {
```

Comment:
Will this break existing use cases? We do this for example:

```
@Singleton
@Primary
final class KievInstantSerde implements Serde<Instant> {
    private final InstantSerde instantSerde;

    public KievInstantSerde(InstantSerde instantSerde) {
        this.instantSerde = instantSerde;
    }

    @Override
    public @Nullable Instant deserialize(@NonNull Decoder decoder, @NonNull DecoderContext context, @NonNull Argument<? super Instant> type) throws IOException {
        if (decoder instanceof KievDecoder kievDecoder) {
            Timestamp timestamp = kievDecoder.decodeTimestamp();
            return timestamp.toInstant();
        } else {
            return instantSerde.deserialize(decoder, context, type);
        }
    }
```

How will this work with this change?

Reply:
Ok this needs to be corrected